chain/consensus/mir/manager.go

```diff
@@ -271,14 +369,25 @@ func (m *Manager) GetMessages(batch []Tx) (msgs []*types.SignedMessage, crossMsg
 	return
 }
 
-func (m *Manager) GetRequests(msgs []*types.SignedMessage, crossMsgs []*types.UnverifiedCrossMsg) (
+func (m *Manager) TransportRequests(msgs []*types.SignedMessage, crossMsgs []*types.UnverifiedCrossMsg) (
 	requests []*mirproto.Request,
 ) {
 	requests = append(requests, m.batchSignedMessages(msgs)...)
 	requests = append(requests, m.batchCrossMessages(crossMsgs)...)
 	return
 }
 
+func (m *Manager) ReconfigurationRequest(payload []byte) *mirproto.Request {
```

Comment:
Question answered :) https://github.com/filecoin-project/eudico/blob/2ab9450496f49e0724cebd18fc16117759aa8544/chain/consensus/mir/mine.go#L57

Reply:
We reconfigure as soon as we get f+1 votes on the reconfiguration request.

`reconfigure ` ticker is how often we check the validator set via the subnet actor. Then we send a special transaction with the new validator set. If it gets f+1 votes then it will be applied in +2 epochs.

Reply:
LGTM. This means that effectively there is no reconfiguration until this ticker checks the validator set, right? (this ticker is basically the epoch I was referring to, we don't want a peer leaving and joining to force continuous reconfigurations DoS'ing the protocol).

Reply:
> This means that effectively there is no reconfiguration until this ticker checks the validator set, right?

That's right.

> we don't want a peer leaving and joining to force continuous reconfigurations DoS'ing the protocol

We probably do not want this. But now any changes in the subnet actor detected by the Mine function trigger reconfiguration in Mir.

Reply:
In fact, it's not quite that.
The ticker only determines the polling interval, at which the node checks whether there were any changes in the validator set. Then everybody "votes" on these changes using the subnet's consensus protocol, and once there is enough votes for the same new configuration, the new configuration is only **scheduled** to be applied at the next reconfiguration occasion (when the `NewConfig` event is triggered in response to `NewEpoch`). If the configuration is changed multiple times in fast succession, then only the most recent one is actually performed.

Reply:
Actually, there might be a little liveness issue with this approach... Under constant and high churn, the nodes might actually never see the same configuration. We only get liveness guarantees here under the assumption that eventually there will be a period of time without changes to the membership that is long enough for a quorum of correct nodes to observe the same configuration.
I think we can live with this assumption for the time being, I'll create an issue for this and we need to come back to it and implement something more sophisticated to guarantee progress even under constant churn.